project/glossary.md

```diff
@@ -0,0 +1,46 @@
+<!-- vim: ft=markdown colorcolumn=72
+-->
+# Glossary
+
+This glossary is a dictionary of terms defined as they are used by the
+spire/helm-charts project.  Writing the definitions of terms here helps
+to keep other documents concise and precise.  Documenting terminology
+here helps prevent misundersandings, facilitating easy onboarding of new
+team members.
+
+**Deployment**
+: A use of the Helm Charts to describe a single SPIRE cluster.
+
+**Deployment Type**
+: One of three supported deployments of the template: Standalone, 
+Primary, or Secondary.
+
+**Federation**
+: When one cluster is configured to trust elements of another cluster
+containing a different trust domain.
+
+**Federated Deployment**
+: A Primary Deployment or Standalone Deployment that is configured
+to federate with one or more Deployments.
+
+**Primary Deployment**
+: A deployment of the Helm Charts where the cluster's purpose is to
+provide certificates to Secondary Deployments.
+
+**Secondary Deployment**
+: A deployment of the Helm Charts where the cluster obtains an
+intermediate CA from a Primary Deployment and uses it to service
+Workload Identity requests.
+
+**Standalone Deployment**
+: A deployment of the Helm Charts where the cluster both manages the CA
+and services Workload Identity requests in the same cluster.
+
+**Tiered Deployment**
+: Use of the Helm Charts two or more times, such that one chart is
```

Comment:
A root chart is likely to contain at least a spire-server and likely a spiffe-oidc-discovery-provider

```suggestion
: Use of the Helm Charts two or more times, such that one or more charts are
```

Reply:
I would prefer to consider "use of the Helm Charts" in a context where one "of the Charts" matches one cluster, not one use "of the charts" match one of each of the possible components in a cluster.  With the latter, undesired, interpretation, to deploy a standalone cluster, one would use the charts two or three times, which just seems an overly complex (and wrong) way to describe things.

Reply:
Oh. I read this differently then, so there is some ambiguity. I took "one chart is configured" to be referring to one of the helm charts (source code), and "the others" referring to other charts (more source code). helm chart refers to code.

Your not talking about charts though, your talking about either what helm calls a Release (instantiation of a chart, https://helm.sh/docs/intro/using_helm/#three-big-concepts) or what this document has been calling a Deployment.

Reply:
Please read the whole sentence, "Use of the Helm Charts two or more times" is not a decorative statement.

Using the word release was intentionally avoided, because we will be re-releasing the Helm Charts for the same product (SPIRE) multiple times.